### ISSUE TITLE
Why editor templates are not being applied?

### ISSUE THREAD

**Issue by JanLenoch** [Kentico/xperience-training-13#53]
### Brief bug description

When using `Html.EditorFor()` with a custom template, shallow models render just fine. However, deeper ones don't. Creating custom templates for all types used in the whole model did not do the trick.

### Repro steps

1. Run the app.
1. Navigate to /identity/account/register/
1. Observe that only `FirstName` and `LastName` properties do get rendered by the `String.cshtml` custom template.
1. Other `string` properties don't as they reside deeper in the model hierarchy.

### Expected behavior

All form fields render by the `String.cshtml` custom editor template, regardless of their depth in the model.

### Test environment

 - Platform/OS: [e.g. .NET Core 3.1, iOS, Windows]
 - Browser [e.g. Chrome, Safari, Edge Chromium, Edge]
 - Version [e.g. 81]

### Additional context

Add any other context about the problem here. If you found an issue where a similar bug has been reported, add a link here.

### Screenshots

Add links to screenshots, if you deem it appropriate.

### Issue expands upon

This issue expands upon another issue(s): #`issue number`

### Blockers

This issue is blocked by the following issue(s): #`issue number`
